Tabloid/client/src/components/CategoryEditForm.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from "react";
+import { FormGroup, Label, Button, Form, Input } from "reactstrap";
+import { editCategory, getCategory } from "../modules/categoryManager";
+import { useHistory, Link, useParams } from "react-router-dom";
+
+export const CategoryEditForm = () => {
+  const { categoryId } = useParams();
+  const [category, setCategory] = useState({
+    id: 1,
+    name: "",
+  });
+
+  const history = useHistory();
+
+  useEffect(() => {
+    if (categoryId) {
+      getCategory(categoryId)
+      .then((res) => { 
+          setCategory(res)
+      })}
+    }, [categoryId])
+
+
+    const edittingCategory = (evt) => {
+        evt.preventDefault();
+
+        const edittedCategory = {
+            id: category.id,
+            name: category.name,
+        };
+        editCategory(categoryId, edittedCategory)
```

Comment:
so the second argument here would just be the 'category' state variable.

Reply:
Yes, after it is assigned to the edittedCategory object.

Reply:
I read this one before the prior one. I see what you mean now. Refactored to pass the state variable instead of assigning a copy first.